lib/token-assert.js

```diff
@@ -77,18 +37,79 @@ TokenAssert.prototype.whitespaceBetween = function(options) {
  * @param {Boolean} [options.disallowNewLine=false]
  */
 TokenAssert.prototype.noWhitespaceBetween = function(options) {
+    options.exactly = 0;
+    this.spacesBetween(options);
+};
+
+/**
+ * Requires to have the whitespace between specified tokens with the provided options.
+ *
+ * @param {Object} options.token
+ * @param {Object} options.nextToken
+ * @param {String} [options.message]
+ * @param {Object} [options.atLeast] At least how many spaces the tokens are apart
+ * @param {Object} [options.atMost] At most how many spaces the tokens are apart
+ * @param {Object} [options.exactly] Exactly how many spaces the tokens are apart
+ * @param {Boolean} [options.disallowNewLine=false]
+ */
+TokenAssert.prototype.spacesBetween = function(options) {
     var token = options.token;
     var nextToken = options.nextToken;
-    if (nextToken.range[0] !== token.range[1] &&
-        (options.disallowNewLine || token.loc.end.line === nextToken.loc.start.line)
-    ) {
-        nextToken.whitespaceBefore = '';
+    var atLeast = options.atLeast;
+    var atMost = options.atMost;
+    var exactly = options.exactly;
+
+    if (!token || !nextToken) {
```

Comment:
This is a hold over from the fact that you can pass in nulls, and it shouldn't cause an error. This is copied from `linesBetween` and I'll consolidate them further in a future followup but won't change behavior for BC.